src/combinatorics/simple_graph/srg.lean

```diff
@@ -0,0 +1,179 @@
+  * The number of common neighbors between any two adjacent vertices in `G` is `l`
+  * The number of common neighbors between any two nonadjacent vertices in `G` is `m`
+
+TODO:
+- Prove that the complement of a strongly regular graph is strongly regular
+
+- Prove that the parameters of a strongly regular graph
+  obey relation `(n - k - 1) * m = k * (k - l - 1)`
+
+- Prove that if I is the identity matrix and J is the all-one matrix,
+  then the adj matrix A of SRG obeys relation A^2 = kI + lA + m(J - I - A)
+
+TODO: Part of this would include defining the complement of a graph
+
+-/
+universes u
+
+namespace simple_graph
+variables {V : Type u} [fintype V]
+variables (G : simple_graph V)
+
+/--
+The set of common neighbors between two vertices `v` and `w` in a graph `G` is the
+intersection of the neighbor sets of `v` and `w`.
+-/
+def common_neighbors (v w : V) : set V := (G.neighbor_set v) ∩ (G.neighbor_set w)
+
+lemma common_neighbors_eq_inter (v w : V) :
+  G.common_neighbors v w = (G.neighbor_set v) ∩ (G.neighbor_set w) := rfl
+
+lemma not_mem_left_common_neighbors (v w : V) : v ∉ (G.common_neighbors v w) :=
+begin
+  rw common_neighbors,
+  simp only [set.mem_inter_eq, irrefl, mem_neighbor_set, not_false_iff, false_and],
+end
+
+lemma not_mem_right_common_neighbors (v w : V) : w ∉ (G.common_neighbors v w) :=
+begin
+  rw common_neighbors,
+  simp,
+end
+
+variables  [locally_finite G]
+
+/--
+A graph is strongly regular with parameters `n k l m` if
+ * its vertex set has cardinality `n`
+ * it is regular with degree `k`
+ * every pair of adjacent vertices has `l` common neighbors
+ * every pair of nonadjacent vertices has `m` common neighbors
+-/
+structure is_SRG_of (n k l m : ℕ) : Prop :=
+(card : fintype.card V = n)
+(regular : G.is_regular_of_degree k)
+(adj_common : ∀ (v w : V), G.adj v w → fintype.card (G.common_neighbors v w) = l)
+(nadj_common : ∀ (v w : V), ¬ G.adj v w → fintype.card (G.common_neighbors v w) = m)
+
+lemma degree_lt_card_verts (G : simple_graph V) (v : V) : G.degree v < fintype.card V :=
+begin
+  classical,
+  rw ← finset.card_univ,
+  rw degree,
+  apply finset.card_lt_card,
+  rw finset.ssubset_iff,
+  use v,
+  split,
+  { simp },
+  { apply finset.subset_univ },
+end
+
+variables [nonempty V]
+
+lemma is_regular_degree_lt_card_verts (G : simple_graph V) (k : ℕ) (h : G.is_regular_of_degree k) :
+  k < fintype.card V :=
+begin
+  rw is_regular_of_degree at h,
+  inhabit V,
+  specialize h (default V),
+  rw ← h,
+  apply degree_lt_card_verts,
+end
+
+lemma common_neighbors_subset_neighbor_set (v w : V) : G.common_neighbors v w ⊆ G.neighbor_set v :=
+begin
+  rw common_neighbors_eq_inter,
+  exact set.inter_subset_left _ _,
+end
+
+lemma set.to_finset_subset_to_finset_iff (α : Type*) (s t : set α) [fintype s] [fintype t] :
+  s ⊆ t ↔ s.to_finset ⊆ t.to_finset :=
+⟨λ h x hx, set.mem_to_finset.mpr $ h $ set.mem_to_finset.mp hx,
+λ h x hx, set.mem_to_finset.mp $ h $ set.mem_to_finset.mpr hx⟩
```

Comment:
I think this should go in `data/set/finite` or wherever `to_finset` is defined

Reply:
and imo the `iff` should be the other direction.